api/STYLE.md

```diff
@@ -141,12 +141,11 @@ To add an extension config to the API, the steps below should be followed:
    (and to not break the docs build).
    See the [key-value-store PR](https://github.com/envoyproxy/envoy/pull/17745/files) for an example of adding a new extension point to common.
 1. Make sure your proto imports the v3 extension config proto (`import "udpa/annotations/status.proto";`)
-1. Make sure your proto is either tracked as a work in progress
-   (`option (udpa.annotations.file_status).work_in_progress = true;`)
-   or ready to be used
+1. Make sure your proto is tracked as ready to be used
    (`option (udpa.annotations.file_status).package_version_status = ACTIVE;`).
-   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD).
-1. Add a reference to the v3 extension config in (1) in [api/versioning/BUILD](versioning/BUILD) under `active_protos`.
+   This is required to automatically include the config proto in [api/versioning/BUILD](versioning/BUILD), and without it `proto_format.sh` will delete the proto file.
+1. Add a reference to the v3 extension config in (2) in [api/versioning/BUILD](versioning/BUILD) under `active_protos`.
+1. Add a reference to the v3 extension config in (2) in [api/BUILD](BUILD) under `v3_protos`. Without this, `proto_format.sh` will erase any imports of your proto file from other proto files.
```

Comment:
Neither of these sound right since both of these files are auto generated, but I haven't looked at this in a while. @phlax @htuch @adisuissa can you advise? @ravenblackx I'm wondering if the problem you are having is that you are not actually consuming your new protos in an actual extension yet, you are just doing a prep/common PR.

Reply:
I agree that the active_protos one seems wrong (that bullet was already in the doc, and I was torn as to whether to remove it!) I had observed that one being auto-generated when I run proto_format.sh.

The v3_protos one doesn't seem to be getting auto-generated, and it's not that it's unused (the 'testing' was in PR20332 which is at least *somewhat* using it in an extension). The other proto file that was trying to import it and getting reverted by proto_format.sh is a prep PR that doesn't use *that* proto yet, but it was the one from PR20332 that lacked the v3_protos entry, which was the cause of that problem.

Reply:
First, I agree this is confusing. What's going on is a form of fixed function computation. Despite the warning on the file and the fact it is automatically generated, it does in fact rely on an initial addition of a new file to allow the tool to converge to the new state. So, the advice of updating `api/BUILD` is correct, we should change the warning at the top of the file to reflect the use here. We could also possibly simplify this tooling significantly further or make its data flow / dependency more explicit.

For the curious, what is going on is that `api/BUILD` is auto-generated, as per https://github.com/envoyproxy/envoy/blob/main/api/BUILD#L1. It is generated by `tools/type_whisperer/proto_build_targets_gen.py`. This relies on the type database as source of truth, https://github.com/envoyproxy/envoy/blob/ba362a22b43a41379b156563f2f83402762081bf/tools/type_whisperer/BUILD#L64. The type database is built from `api/BUILD` (https://github.com/envoyproxy/envoy/blob/ba362a22b43a41379b156563f2f83402762081bf/tools/type_whisperer/BUILD#L59).

Reply:
I agree that the simple way forward is clarifying the comment here:
https://github.com/envoyproxy/envoy/blob/2e9179f6ab7fe044fc3d9419452251101e4b6cc8/tools/type_whisperer/proto_build_targets_gen.py#L39

An alternative approach could be traversing the proto files, and creating a list of v2_deps and v3_deps.

Reply:
Agree that it would be best for the tooling to *actually* do this automatically, and second-best to update the DO NOT EDIT in addition to STYLE.md.
As a short-term fix to just make it *less* confusing until either one of those things happens, the proposed additional bullet in STYLE.md is sufficient to at least make it a 5 minute confusion instead of a 2 hour confusion. I'd rather not expand this change out into also touching .py files.